### ISSUE TITLE
Add Mahalanobis OOD

### ISSUE THREAD

**Issue by pwhofman** [pwhofman/probly#512]
We should add the following method to probly: Mahalanobis OOD --- a post-hoc OOD detector applicable to any pre-trained softmax classifier.
- Paper: https://arxiv.org/abs/1807.03888;
- Reference implementation: https://github.com/pokaxpoka/deep_Mahalanobis_detector.
